evalml/tests/component_tests/test_undersampler.py

```diff
@@ -74,3 +75,55 @@ def test_undersample_imbalanced(data_type, make_data_type):
 
     np.testing.assert_equal(X, transform_X.to_dataframe().values)
     np.testing.assert_equal(None, transform_y)
+
```

Comment:
Can you add coverage the random seed is preserved when users pass in `sampling_dict`? Your implementation does that now but just suggesting so that we preserve that behavior during a refactor, especially since we init the component during `fit` and not `__init__`